oidc/transport.go

```diff
@@ -47,24 +47,24 @@ type AuthenticatedTransport struct {
 	http.RoundTripper
 
 	mu  sync.Mutex
```

Comment:
Comment from slack:

TokenRefresher is an interface. why not this?

```
type tokenRefreshWithInitialJWT struct {
    jwt       jose.JWT
    once      sync.Once
    refresher oidc.TokenRefresher
}

func (t *tokenRefreshWithInitialJWT) Verify(jwt jose.JWT) error {
    return t.refresher.Verify(jwt)
}

func (t *tokenRefreshWithInitialJWT) Refresh() (jwt jose.JWT, err error) {
    // Only set the initial token once.
    t.once.Do(func() { jwt = t.jwt })
    if jwt != nil {
        return jwt, nil
    }
    return t.refresher.Refresh()
}
```

getter/setter works fine too if you feel it's a common enough use case

or maybe a constructor that sets the private fields